### ISSUE TITLE
Great work. I think this issue is the biggest to fix.

### ISSUE THREAD

**Issue by NathanDrake2406** [Nukem9/dlssg-to-fsr3#234]
I think you're already aware of it: UI and HUD ghosting.

I managed to make it work for almost every game I tested :) From games that have no native DLSS FG support, like TLOU remake using Puredark's mod, to games that have native FSR 3 (Avatar). I turned off FSR 3 native, mod DLSS FG into that game and then mod that FG into FSR3. DLSS upscale + FSR FG is nice! It would be near perfect if the UI and HUD issue is fixed.

**Comment by Dravic** [comment]
I'll be honest: the biggest issues are the consistent crashes caused by the mod in various games, not the ghosting.

**Comment by Nooblet16** [comment]
> I think you're already aware of it: UI and HUD ghosting.
> 
> I managed to make it work for almost every game I tested :) From games that have no native DLSS FG support, like TLOU remake using Puredark's mod, to games that have native FSR 3 (Avatar). I turned off FSR 3 native, mod DLSS FG into that game and then mod that FG into FSR3. DLSS upscale + FSR FG is nice! It would be near perfect if the UI and HUD issue is fixed.

The UI issues are nothing to do with mod, that's just how the game solves UI issues with framegen. DLSS has it too and there's really no proper way around it.

The only way to fix UI issues when doing framegen is to have them run at the original framerate otherwise you get garbling. It's just a bit more vissible in Avatar's case due to how its UI is designed in the first place i.e. it has a clear border surrounding the text that is much larger than the text and that border gets affected by this half rate too. The way to fix it would be to no thave such large transparent borders around the UI.

**Comment by Dravic** [comment]
>DLSS has it too and there's really no proper way around it.

Incorrect. DLSS has different ways of dealing with UI interpolation. They actually, in my experience, most often brute force it with a healthy dose of UI masking and some ML magic.

DLSS3 in games like Diablo 3 or Cyberpunk 2077 straight up interpolates the UI. This way there's no "stuttering", but occasionally there can be these split second UI garbling artifacts. I find this way more pleasant than FSR3's often used half framerate UI update which looks like stuttering.

**Comment by buraktsgs** [comment]
> > I think you're already aware of it: UI and HUD ghosting.
> > I managed to make it work for almost every game I tested :) From games that have no native DLSS FG support, like TLOU remake using Puredark's mod, to games that have native FSR 3 (Avatar). I turned off FSR 3 native, mod DLSS FG into that game and then mod that FG into FSR3. DLSS upscale + FSR FG is nice! It would be near perfect if the UI and HUD issue is fixed.
> 
> That UI issue has nothing to do with mod, that's just how the game devs chose to solve UI issues with framegen. DLSS has it too and there's really no proper way around it.
> 
> The only way to fix UI issues when doing framegen is to have them run at the original framerate otherwise you get garbling. It's just a bit more vissible in Avatar's case due to how its UI is designed in the first place i.e. it has a clear border surrounding the text that is much larger than the text and that border gets affected by this half rate too. The way to fix it would be to no thave such large transparent borders around the UI.

I dont know if its possible but I think we will see custom frame gen ui mods for games in the future.

**Comment by NathanDrake2406** [comment]
> I'll be honest: the biggest issues are the consistent crashes caused by the mod in various games, not the ghosting.

Haven't crashed once for me. Tested in games that support FG natively and games that need to add FG to it.

**Comment by Dravic** [comment]
> > I'll be honest: the biggest issues are the consistent crashes caused by the mod in various games, not the ghosting.
> 
> Haven't crashed once for me. Tested in games that support FG natively and games that need to add FG to it.

Hogwart's Legacy crashes frequently.

In The Witcher 3 you can simply spin the camera fast and a few seconds later you'll crash.

**Comment by NathanDrake2406** [comment]
> > > I'll be honest: the biggest issues are the consistent crashes caused by the mod in various games, not the ghosting.
> > 
> > 
> > Haven't crashed once for me. Tested in games that support FG natively and games that need to add FG to it.
> 
> Some examples:
> 
> Hogwart's Legacy crashes frequently.
> 
> In The Witcher 3 you can simply spin the camera fast and a few seconds later you'll crash.

Eh, I had a 2 hour session of Witcher 3 yesterday using the mod. The game is truly beautiful because, thanks to this mod, I was able to max out RT. No crashes. 
I also tested Hogwarts Legacy for 15-20 mins, no crashes.

**Comment by Z4ndyz** [comment]
I wonder if nukem can try to use a method of UI interpolation in fsr3, i havent actually take a deep look in all of fsr3's source code but it would be an interesting option to try

**Comment by Sousaboi** [comment]
I too am facing the issue as well of the interpolation..
